src/core/Context.py

```diff
@@ -64,32 +69,48 @@ async def prompt(self, message, title=None, delete_after=True):
             embed.title = title
 
         view = utils.Prompt(self.author.id)
-        msg = await self.send(embed=embed, view=view)
+        msg: Optional[discord.Message] = await self.send(embed=embed, view=view)
         await view.wait()
 
         try:
-            if delete_after:
+            if delete_after and msg is not None:
                 await msg.delete(delay=0)
         finally:
             return view.value
 
-    async def error(self, message, delete_after=None, **kwargs) -> discord.Message:
+    async def error(
+        self, message: str, delete_after: bool = None, **kwargs: Any
+    ) -> Optional[discord.Message]:
         with suppress(discord.HTTPException):
 
-            return await self.reply(
+            msg: Optional[discord.Message] = await self.reply(
                 embed=discord.Embed(description=message, color=discord.Color.red()),
                 delete_after=delete_after,
                 **kwargs,
             )
+            try:
+                await self.bot.wait_for(
+                    "message_delete", check=lambda m: m.id == self.message.id, timeout=30
+                )
+            except asyncio.TimeoutError:
+                pass
+            else:
+                if msg is not None:
+                    await msg.delete(delay=0)
+            finally:
+                return msg
 
-    async def safe_delete(self, msg: discord.Message, sleep_for: int = 0):
+        return None
+
+    async def safe_delete(self, msg: discord.Message, sleep_for: Union[int, float] = 0) -> None:
         if sleep_for:
             await asyncio.sleep(sleep_for)
 
-        with suppress(discord.HTTPException, AttributeError):
-            await msg.delete()
+        await msg.delete(delay=0)
```

Comment:
`with suppress(...):` makes the code more pythonic, no doubt, but the context-manager always makes your codes little bit slow. As you used the `contextlib` module in almost every cog, I should say it makes the code base slow.

Consider using `try-except-else-finally`. It won't affect the speed at all.

`contextlib` should be used in more often cases, like File handling, or excepting the obvious errors like `IndexError`

`try-except` is used when we make proper error handling

Reply:
actually `contextlib.suppress()` is merely a convenient tool for me, yes it might be affecting the speed of execution but the pretty tidy syntax can't be neglected as well, 
and since Quotient is being used a lot these days maybe this is high time to start working on optimizing & refactoring the whole codebase, thanks.